### PR TITLE
Don't show sources if their count has not changed

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Description

Follow-up to #4067

Here's the report from the most recent audio data refresh:

image

The current logic is set to show all sources, regardless of whether there was a change or not. This PR changes the logic so only sources that have increased or decreased in size are shown in the final report. While this doesn't have a huge effect on audio data refreshes, it will for image data refreshes since we have significantly more sources there.

## Testing Instructions

Updated tests should pass. Alternatively, run the data refresh locally after `just api/init` and see that no sources are listed in the completion message because none of the counts changed. 

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: catalog/tests/dags/data_refresh/test_reporting.py [hunk 1]
```diff
@@ -19,7 +19,7 @@
         [
             {"src1": 4, "src2": 21},
             {"src1": 4},
-            ["25 → 4", "-21 (-84.0%", "`src1`:+0", "`src2`:-21"],
+            ["25 → 4", "-21 (-84.0%", "`src2`:-21"],
```
Comment (krysal) [inline catalog/tests/dags/data_refresh/test_reporting.py:22]: It would be nice to add a note that unchanged sources are omitted.

**Review by krysal — APPROVED** [review]
Nice.

**Review by dhruvkb — COMMENTED** [review]
+1 for @krysal's suggestions. Do we need to handle a case where all sources are unchanged? That seems unlikely to ever occur at all.

**Comment by AetherUnbound** [comment]
> Do we need to handle a case where all sources are unchanged? That seems unlikely to ever occur at all.

Only in local testing I would imagine! I don't think it's worth having logic for that case, personally. And I've made the message about unchanged sources ambiguous enough that it should cover the case where all sources are absent too

**Review by obulat — APPROVED** [review]
LGTM!

**Pull request merged**